rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -526,8 +550,11 @@ impl TunnelManager {
     pub fn neighbor_inquiry_hostname(&mut self, their_hostname: String) -> Result<(), Error> {
         trace!("Getting tunnel, inq");
 
-        let our_port = match self.free_ports.pop() {
-            Some(p) => p,
+        let our_port = match self.port_query() {
+            Some(port) => {
+                self.ports.insert(port, false);
```

Comment:
why no have port_query mark the port it gives you as not free?

Reply:
Borrowing `self.ports` as mutable would conflict with `is_free`'s immutability (as a `bool`). Then, even if I make a mutable reference when borrowing, there can only be one mutable reference to `self.ports` so the borrow checker would be violated if I `self.ports.insert(port, false)` in the same scope. Though, I'm still learning Rust - there's probably a way.